docs/src/operations/upgrade.md

```diff
@@ -20,11 +20,117 @@ The following procedure describes how to upgrade a broker. If the cluster contai
 
 Note that this procedure results in a downtime of the whole cluster. Currently, a rolling upgrade is not recommended to avoid that an upgraded broker do reprocessing and is affected by the issue.
 
-> TODO: describe how to perform the upgrade on Kubernetes/Helm
+### Upgrade in Kubernetes
+#### Preparing the Upgrade
+##### From broker version < 0.24.4
 
-### Preparing the Upgrade
+1. Stop the workflow processing
+    * Close all job workers
+    * Interrupt the incoming connections to avoid user commands
+1. Wait until a snapshot is created for all partitions
+    * By default, a snapshot is created every 15 minutes
+    * Verify that a snapshot is created by looking at the [Metric](/operations/metrics.md) `zeebe_snapshot_count` on the leader and the followers
+    * Note that no snapshot is created if no processing happened since the last snapshot
+1. Make a backup of the pvc's
+    * If you're using Kubernetes 1.17+, then there is a new beta feature, which you can use to snapshot your pvc. Check [this documentation](https://kubernetes.io/docs/concepts/storage/volume-snapshots/) for more information.
+    * If you're using an older Kubernetes version then you need to create snapshot's of your pvc's with help of your cloud provider.  For example for the google cloud kubernetes engine follow [this guide](https://cloud.google.com/compute/docs/disks/create-snapshots).
+
+##### From broker version >= 0.24.4
+
+The [Partitions Admin Endpoint](#partitions-admin-endpoint) can be used to query the status of the partitions and perform operations for the upgrade.
+
+First we need stop the workflow processing and trigger a snapshot creation.
+In order to do that we need to call the `prepareUpgrade` operation on the partitions endpoint.
+You need to port-forward to each Zeebe pod and send a POST request to the endpoint.
+
+**Example:**
+Forward traffic to the pod's metric port (default 9600).
+```shell script
+$ kubectl port-forward PODNAME 9600
+Forwarding from 127.0.0.1:9600 -> 9600
+Forwarding from [::1]:9600 -> 9600
+```
+
+Send post request to the `prepareUpgrade` endpoint.
+```shell script
+$ curl -X POST localhost:9600/actuator/partitions/prepareUpgrade
+{"1":{"role":"FOLLOWER","snapshotId":null,"processedPosition":null,"processedPositionInSnapshot":null,"streamProcessorPhase":null},"2":{"role":"FOLLOWER","snapshotId":null,"processedPosition":null,"processedPositionInSnapshot":null,"streamProcessorPhase":null},"3":{"role":"LEADER","snapshotId":null,"processedPosition":-1,"processedPositionInSnapshot":null,"streamProcessorPhase":"PAUSED"}}
+```
+
+**You need to do this for all pods (brokers) to make sure that you paused all partition leaders.**
```

Comment:
Having this line here makes it sound like only the querying needs to be done on the leaders, when we need to do everything on the leaders. Maybe put it higher, or start this paragraph with this? I would also add how to find the nodes that are leaders for which partition (e.g. topology), and that the query below can help them verify this information.

Reply:
The idea was that we just do it on all nodes, such that we don't need to find the leaders. And when you query the endpoint you see the leaders of the node.

Reply:
Maybe we can pull this up so users are aware of this (doing it on all pods) before we give the steps? So something like

```md
##### From broker version >= 0.24.4

The [Partitions Admin Endpoint](#partitions-admin-endpoint) can be used to query the status of the partitions and perform operations for the upgrade. First we need stop the workflow processing, trigger a snapshot creation, then query the broker status repeatedly to ensure all conditions are met.

To do so, you need to do the following **for all brokers in your cluster**. If you are using the Helm chart, you can find all broker pods by running the following command: `kubectl get pods -l app.kubernetes.io/component=broker`.

Forward traffic to the broker's metric port (default 9600). This will allow you to send local requests via `curl` to the broker's admin endpoint.

```shell script
$ kubectl port-forward PODNAME 9600
Forwarding from 127.0.0.1:9600 -> 9600
Forwarding from [::1]:9600 -> 9600
```

Send post request to the `prepareUpgrade` endpoint. This will pause stream processing and will trigger snapshot creation.

```shell script
$ curl -X POST localhost:9600/actuator/partitions/prepareUpgrade
{"1":{"role":"FOLLOWER","snapshotId":null,"processedPosition":null,"processedPositionInSnapshot":null,"streamProcessorPhase":null},"2":{"role":"FOLLOWER","snapshotId":null,"processedPosition":null,"processedPositionInSnapshot":null,"streamProcessorPhase":null},"3":{"role":"LEADER","snapshotId":null,"processedPosition":-1,"processedPositionInSnapshot":null,"streamProcessorPhase":"PAUSED"}}
```

Then, query the status of the broker's partitions repeatedly using the following command:
```shell script
$ curl -X GET localhost:9600/actuator/partitions
{"1":{"role":"FOLLOWER","snapshotId":null,"processedPosition":null,"processedPositionInSnapshot":null,"streamProcessorPhase":null},"2":{"role":"FOLLOWER","snapshotId":null,"processedPosition":null,"processedPositionInSnapshot":null,"streamProcessorPhase":null},"3":{"role":"LEADER","snapshotId":null,"processedPosition":-1,"processedPositionInSnapshot":null,"streamProcessorPhase":"PAUSED"}}
```

until the following conditions are met for all partitions:
  * `processedPositionInSnapshot` is equal to `processedPosition`
  * all followers (partitions which report `"role": "FOLLOWER"`) have the same `snapshotId` as their leader (partitions which report `"role": "LEADER"`)

As a reminder, for a partition which is `FOLLOWER` on a given broker, another broker will be the `LEADER` for that same partition. To match that both leaders and followers have the same snapshot, you may need to query different brokers to get all the information.
```

I have a feeling I didn't write it so well, but you get the idea - I think it's helpful for users if we explain each step we're doing, and that the FOLLOWER/LEADER stuff may be spread across different brokers.